local_db/api.py

```diff
@@ -95,3 +129,36 @@ def set_status(self, request: ReleaseRequest, status: Status, user: User):
             metadata.status = status
             metadata.save()
             super().set_status(request, status, user)
+
+    def add_file_to_request(
+        self,
+        release_request: ReleaseRequest,
+        relpath: Path,
+        user: User,
+        group_name: Optional[str] = "default",
+    ) -> ReleaseRequest:
+        # call super() to copy the file
+        super().add_file_to_request(release_request, relpath, user, group_name)
```

Comment:
So, thinking this through, if the copy succeeds but the db fails, we are left with inconsistent state.

So it might be better to write to the db first, and then call super() to copy *within the transaction block*.

That way, if the copy fails, the db rolls back. And if the db fails, the copy never happens. In either case, we are left with consistent state.

Might need to update the doc string on ProviderAPI.add_file_to_request to match that if so, I think I got it wrong by saying it should call it first.

Reply:
Yes, that makes sense, and now that I've started poking things from the frontend I've come across that inconsistent state already